en/docs/guides/authentication/mobile-app.md

```diff
@@ -1,33 +1,82 @@
-# Enable Authentication to a mobile application using Authorization code grant with PKCE
+# Enable Authentication for a Mobile Application
 
-< Intro - Step by Step guide to enable authentication for a
-mobile/native application>
+This page guides you through enabling authentication using the [Authorization Code](insertlink) grant type with [PKCE](insertlink) for a mobile application that uses OpenID Connect. 
 
-## Overview
- < Explain the use case with diagrams - can be links to concepts > 
- 
- < Image explaining the scenario>
- 
-## Register Application
+---
 
- < Description on mobile app specific config >
+This guide assumes you have your own application. If you wish to try out this flow with a sample application, click the button below. 
+
+<a class="samplebtn_a" href="../../../samples/regular-webapp-oidc-sample" target="_blank" rel="nofollow noopener">Try it with the sample</a>
+
+----
+
+{!fragments/register-a-service-provider.md!}
+
+----
 
 {!fragments/oauth-app-config-basic.md!}
 
-| Field                 | Value         | 
-| --------------------- | ------------- | 
-| Service Provider Name | sample- app  |
-| Description           | This is a mobile application  | 
-| Call Back Url         | https://somehost/url  | 
+----
+
+
+## Enable PKCE
+
+When configuring **OAuth/OpenID Connect Configuration** for the service provider, select **PKCE Mandatory** in order to enable PKCE. 
+
+![enable-pkce](../../assets/img/guides/enable-pkce.png)
+
+----
 
-{!guides/authentication/oauth-app-config-advanced.md!}
+## Configure the client application
 
-{!guides/authentication/configure-mobile-app.md!}
+Make the following requests via your application to connect your application to WSO2 IS. 
 
-{!fragments/get-user-information-to-the-oidc-application.md!}
+1. Obtain the `authorization_code` by sending an authorization request to the authorization endpoint. 
+
+    !!! tip
+        You can use [an online tool](https://tonyxu-io.github.io/pkce-generator/) to generate PKCE code challenges to include the `code challenge` and `code_challenge_method` parameters. 
+
+    ```tab="Request Format"
+    https://<host>/oauth2/authorize?scope=openid&response_type=code
+    &redirect_uri=<redirect_uri>
+    &client_id=<OAuth Client Key>
+    &code_challenge=<PKCE_code_challenge>
+    &code_challenge_method=<PKCE_code_challenge_method>
+    ```
+
+    ```tab="Sample Request"
+    https://localhost:9443/oauth2/authorize?scope=openid&response_type=code
+    &redirect_uri=https://localhost/callback
+    &client_id=YYVdAL3lLcmrubZ2IkflCAuLwk0a
+    &code_challenge=5vEtIy2T-G65yXHc8g5zcJDQXICBzZMrtERq0zhx7hM
+    &code_challenge_method=S256
+    ```
+
+2. Obtain the access token by sending a token request to the token endpoint using the `authorization_code` recieved in step 1, and the `OAuth Client Key` and `<OAuth Client Secret>` obtained when configuring the service provider.
```

Comment:
`OAuth Client Key` and `<OAuth Client Secret>` are inconsistent. Either let's use `<OAuth Client Key>` and `<OAuth Client Secret>` or `OAuth Client Key` and `OAuth Client Secret`